packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js

```diff
@@ -46,29 +46,26 @@ function calculateSplitIndex({
   isFirstRow,
   fitsFirstRow,
   availableSlotsOnFirstRow,
-  columnCount,
 }: {
   oneColumnItemsLength: number,
   twoColumnIndex: number,
   isFirstRow: boolean,
   fitsFirstRow: boolean,
   availableSlotsOnFirstRow: number,
-  columnCount: number,
 }): number {
   // We add one more item to pre so it is positioned instead of two column item
-  if (isFirstRow && twoColumnIndex < columnCount && !fitsFirstRow) {
+  if (isFirstRow && twoColumnIndex < availableSlotsOnFirstRow && !fitsFirstRow) {
```

Comment:
slightly confused here.  `availableSlotsOnFirstRow` should be the number of columns that we have available to fit a module in right? wouldn't it make sense to still compare `index` to total count? 

e.g.if we have 6 column grid with two columns available in the first row and the two column module is in index 3, we would be returning 2 here as the split index